### PR TITLE
feat: add maximum maneuver duty cycle constraint

### PR DESCRIPTION
**Pull request by phc1990**
Adds maximum maneuver duty cycle constraint. Expressed as "time over time", e.g. {40 min, 98min} meaning as much as 40 min of burn for any 98 min interval).

## Summary by CodeRabbit

* **New Features**
  * Added duty-cycle constraint support for maneuver planning (numerator/denominator pairs) with validation and reporting.
  * Solver now accounts for duty-cycle limits when scheduling maneuvers, can recommend or enforce coast times, and uses a numeric fallback for tightly constrained cases.
  * Maneuver constraint display updated to include duty-cycle state.

* **Tests**
  * Expanded coverage for duty-cycle scenarios, interval validation, boundary/edge cases, and invalid-interval error paths.

### CODE REVIEW

File: include/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.hpp [hunk 1]
```diff
@@ -101,12 +104,41 @@ class Segment
...
+        ///
+        /// @param aMaximumDutyCycle The maximum duty cycle as numerator and denumerator. For example,
+        /// {Duration::Minutes(40.0), Duration::Minutes(98.0)} represents a maximum maneuvering time of 40 minutes over
+        /// any 98 minutes interval.
+        /// @param aMinimumDuration The minimum duration for a maneuver. Defaults to Undefined.
+        /// @param aMaximumDuration The maximum duration for a maneuver. Defaults to Undefined.
+        /// @param aMinimumSeparation The minimum separation between maneuvers. Defaults to Undefined.
+        /// @param aMaximumDurationStrategy The strategy when maximum duration is violated. Defaults to Fail.
+        /// @return ManeuverConstraints instance
+        static ManeuverConstraints ManeuverDutyCycle(
```
Comment (apaletta3) [inline include/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.hpp]: Nit: Perhaps we should wait decide on how best we want to do this breaking change factory method pattern (i.e. naming convention) before introducing it? We can simply make another regular constructor with the 5th argument for the time being, right?

File: include/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.hpp [hunk 2]
```diff
@@ -105,12 +104,39 @@ class Segment
             const Duration& aMinimumSeparation = Duration::Undefined(),
             const MaximumManeuverDurationViolationStrategy& aMaximumDurationStrategy =
                 MaximumManeuverDurationViolationStrategy::Fail
+        )
+            : ManeuverConstraints(
```
Comment (vishwa2710) [inline include/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.hpp]: Let's define this in the cpp rather the hpp.

File: src/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.cpp [hunk 3]
```diff
@@ -868,12 +982,210 @@ Segment::Solution Segment::solve(
...
+
+                        return solveAndAcceptCoast(candidateManeuverInterval.getEnd());
+                    }
+
+                    newDuration -= stepSize;
+                }
+
+                // Could not find a valid maneuver interval, here we can't skip to the end of the maneuver,
+                // the fact that we could not allocate a chunk at the beginning of the maneuver interval does
+                // not mean that we can't allocate a chunk later.
```
[the remark this answers is not in the document]
Comment (phc1990) [reply src/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.cpp]: changed docs, should be correct and clearer now (it was not correct before)

File: include/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.hpp [hunk 4]
```diff
@@ -107,10 +110,28 @@ class Segment
...
+        /// @param aMaximumDurationStrategy The strategy when maximum duration is violated.
+        /// @param aMaximumDutyCycle The maximum duty cycle as numerator and denominator. For example,
+        ///     {Duration::Minutes(40.0), Duration::Minutes(98.0)} represents a maximum maneuvering time of 40 minutes
+        ///     over any 98 minutes interval.
+        ManeuverConstraints(
+            const Duration& aMinimumDuration,
+            const Duration& aMaximumDuration,
+            const Duration& aMinimumSeparation,
+            const MaximumManeuverDurationViolationStrategy& aMaximumDurationStrategy,
+            const Pair<Duration, Duration>& aMaximumDutyCycle
```
Comment (apaletta3) [inline include/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.hpp]: Shouldn't we have done this to make the user's life easier?
```suggestion
            const Duration& aMinimumDuration = Duration::Undefined(),
            const Duration& aMaximumDuration = Duration::Undefined(),
            const Duration& aMinimumSeparation = Duration::Undefined(),
            const MaximumManeuverDurationViolationStrategy& aMaximumDurationStrategy =
                MaximumManeuverDurationViolationStrategy::Fail,
            const Pair<Duration, Duration>& aMaximumDutyCycle = {Duration::Undefined(),  Duration::Undefined()}
```
